File: algorithms/[350]两个数组的交集 II/solution.py
# topics = ["数组", "哈希表"]

from typing import List


class Solution:
    def intersect(self, nums1: List[int], nums2: List[int]) -> List[int]:
        result: List[int] = []
        for num in nums1:
            if num in nums2:
                nums2.remove(num)
                result.append(num)
        return result

File: algorithms/[350]两个数组的交集 II/solution.ts
function intersect(nums1: number[], nums2: number[]): number[] {
  const map = new Map<number, number>();
  for (const ele of nums1) {
    if (map.has(ele)) {
      map.set(ele, map.get(ele)! + 1);
    } else {
      map.set(ele, 1);
    }
  }
  const res: number[] = [];
  for (const ele of nums2) {
    if (map.has(ele)) {
      const counts = map.get(ele);
      if (counts === 1) {
        map.delete(ele);
      } else {
        map.set(ele, map.get(ele)! - 1);
      }
      res.push(ele);
    }
  }
  return res;
};
